goal_src/engine/game/collectables-part.gc

```diff
@@ -5,3 +5,3405 @@
 ;; name in dgo: collectables-part
 ;; dgos: GAME, ENGINE
 
+;; why is this file built first....
+;; TODO - also for entity and beach-rocks
+(deftype collectable (process-drawable)
```

Comment:
We shouldn't move the `deftype`s around - these actually generate code that will allocate the type record and add stuff to the symbol table, and we want to keep that stuff in the same order.  Is it possible to replace these with something like 
```
(declare-type collectable process-drawable)
(declare-type eco-collectable collectable)
```
?

Reply:
i can double check, but not at this time i think -- I believe it complained about it not being fully defined.

Reply:
```
-- Type Error! --
Type eco-collectable is not fully defined.
Code:
((lambda :inline-only #t (v1-1)
 (let ((a0-3 (-> v1-1 root-override root-prim prim-core)))
  (set! (-> arg2 x) (-> a0-3 world-sphere x))
  (set! (-> arg2 y) (-> a0-3 world-sphere y))
  (set! (-> arg2 z) (-> a0-3 world-sphere z))
  )
 (if (zero? (logand (-> v1-1 state-flags) 1))
  (set! (-> arg1 next-time) (the-as uint (* (-> *sp-frame-time* x) 2)))
  )
 )
  (the-as eco-collectable (-> arg1 key proc))
  )
From macro: let
Used in:
goal_src/engine/game/collectables-part.gc, line: 89
```